AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -300,25 +317,36 @@ - (void)deleteDocumentWithPartition:(NSString *)partition
                          documentId:(NSString *)documentId
                        writeOptions:(MSWriteOptions *_Nullable)__unused writeOptions
                   completionHandler:(MSDataSourceErrorCompletionHandler)completionHandler {
-  [self performOperationForPartition:partition
-                          documentId:documentId
-                          httpMethod:kMSHttpMethodDelete
-                                body:[NSData data]
-                   additionalHeaders:nil
-                   completionHandler:^(NSData *_Nullable __unused responseBody, NSHTTPURLResponse *_Nullable __unused response,
-                                       NSError *_Nullable cosmosDbError) {
-                     // Body returned from call (data) is empty.
-                     NSInteger httpStatusCode = [MSDataSourceError errorCodeFromError:cosmosDbError];
-                     if (httpStatusCode != MSHTTPCodesNo204NoContent) {
-                       MSLogError([MSDataStore logTag],
-                                  @"Not able to delete document. Error: %@; HTTP status code: %ld; "
-                                  @"Document: %@/%@",
-                                  cosmosDbError.localizedDescription, (long)httpStatusCode, partition, documentId);
-                     } else {
-                       MSLogDebug([MSDataStore logTag], @"Document deleted: %@/%@", partition, documentId);
-                     }
-                     completionHandler([[MSDataSourceError alloc] initWithError:cosmosDbError]);
-                   }];
+
+  @synchronized(self) {
+    if (![self canBeUsed] || ![self isEnabled]) {
+      NSError *error = [[NSError alloc] initWithDomain:kMSACErrorDomain
+                                                  code:MSACDisabledErrorCode
+                                              userInfo:@{NSLocalizedDescriptionKey : kMSACDisabledErrorDesc}];
+      MSLogError([MSDataStore logTag], @"Not able to delete the document ID:%@ with error:%@", documentId, [error description]);
```

Comment:
Can the NSError and log be extracted into a method? So the method will return the `NSError` and the methods will just invoke the completion handler after that.

Reply:
Seems like it would be a bit more flexible

Reply:
Fair, refactored that helper method.